paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java

```diff
@@ -157,25 +148,22 @@ public boolean equals(Object o) {
                 return false;
             }
             ManifestsSplit that = (ManifestsSplit) o;
-            return Objects.equals(location, that.location);
+            return Objects.equals(rowCount, that.rowCount);
```

Comment:
I think this is dangerous. If I put `ManifestsSplit` in a `HashSet`. The splits those have the same rowCount will be de-duplicated.

Reply:
There is no scenario that requires splitting different tables into one hashmap.